Add stm32h7 dmamux1 bindings

shlomow opened the pull request:
This pr is a preparation for enabling the dma controller for stm32h7 soc.
It adds the dmamux1 bindings and fix the bindings for dma1 and dma2

shlomow committed:
dts: st: h7: fix clocks of dma1 and dma2

Fix the clocks bindings of dma1 and dma2
for stm32h7 series. This is wrong probably
due to copy and paste.

Signed-off-by: Shlomi Vaknin <shlomi.39sd@gmail.com>

shlomow commented:
@erwango

FRASTM commented on dts/arm/st/h7/stm32h7.dtsi:
```diff
@@ -712,6 +712,19 @@
+			#dma-cells = <4>;
+			reg = <0x40020800 0x400>;
+			interrupts = <102 0>;
+			clocks = <&rcc STM32_CLOCK_BUS_AHB1 0x00000001>;
```
Could we have a comment to explain that DMAMUX has no specified clock on the AHB

FRASTM commented on dts/arm/st/h7/stm32h7.dtsi:
```diff
@@ -712,6 +712,19 @@
+			clocks = <&rcc STM32_CLOCK_BUS_AHB1 0x00000001>;
+			dma-channels = <16>;
+			dma-generators = <8>;
+			dma-requests= <107>;
```
not all stm32h7 have 107 dma-request, for example stm32h72x have 129

shlomow committed:
dts: st: h7: add dmamux1 bindings

Add dmamux1 dts bindings for stm32h7
series. Note that there is no dedicated
clock to enable for dmamux1 so we set the
clock of dma1. This is the way linux handle
this.

Signed-off-by: Shlomi Vaknin <shlomi.39sd@gmail.com>
